ui/pages/permissions-connect/connect-page/utils.ts

```diff
@@ -101,11 +111,25 @@ export function getDefaultAccounts(
     }
   });
 
+  if (
+    existingPermittedCaipAccountIds &&
+    existingPermittedCaipAccountIds.length > 0
+  ) {
+    const accounts = allAccounts.filter((account) =>
+      isCaipAccountIdInPermittedAccountIds(
+        account.caipAccountId,
+        existingPermittedCaipAccountIds,
+      ),
+    );
+
+    defaultAccounts.push(...accounts);
+  }
+
   const unsatisfiedNamespaces = requestedNamespaces.filter(
     (namespace) => !satisfiedNamespaces.has(namespace),
   );
 
-  if (unsatisfiedNamespaces.length > 0) {
+  if (unsatisfiedNamespaces.length > 0 && !isRequestingSpecificEvmAccounts) {
```

Comment:
I'm not sure I follow this

Reply:
Yeah I'm not understanding why this resolves @jiexi 's change req:
"If the dapp requests an account and it exists, then we shouldn't also include the currently selected account"

if the dapp requested account exists shouldn't it already be pushed in the defaultAccounts array above and the namespace be satisfied so we wouldn't enter this if block anyways?  🤔 

In any case I would prefer to avoid logic highly specific to EVM like this if possible

Reply:
Hmm I think there's something wrong with a part of this logic -> `supportedRequestedAccounts` is always empty for EVM requests, regardless of specifying accounts or not.

Also agree with the specific EVM logic part.